modules/persistent-storage-csi-azure-file-cross-sub-dynamic-pre-provisioning-pv-pvc-procedure.adoc

```diff
@@ -0,0 +1,87 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent_storage-csi-azure-file.adoc
+//
+:_mod-docs-content-type: PROCEDURE
+[id="persistent-storage-csi-azure-file-cross-sub-dynamic-pre-provisioning-pv-pvc-procedure_{context}"]
+= Static provisioning across subscriptions for Azure File by creating a PV and PVC:
+
+.Prerequisites
+* Installed {product-title} cluster on Azure with the service principal or managed identity as an Azure identity in one subscription (call it Subscription A)
+
+* Access to another subscription (call it Subscription B) with the storage that is in the same tenant as the cluster
+
+* Logged in to the Azure CLI
+
+.Procedure
+. For your Azure File share, record the resource group, storage account, storage account key, and Azure File name. These values are used for the next steps.
+
+. Create a secret for the persistent volume parameter `spec.csi.nodeStageSecretRef.name` by running the following command:
++
+[source, terminal]
+----
+$ oc create secret generic azure-storage-account-{storageaccount-name}-secret --from-literal=azurestorageaccountname="<azure-storage-account-name>" --from-literal azurestorageaccountkey="<azure-storage-account-key>" --type=Opaque
+----
++
+Where:
+`<azure-storage-account-name>` and `<azure-storage-account-key>` are the Azure storage account name and key respectively that you recorded in Step 1.
+
+. Create a persistent volume (PV) by using a similar configuration to the following example file:
++
+.Example PV YAML file
+[source,terminal]
+----
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  annotations:
+    pv.kubernetes.io/provisioned-by: file.csi.azure.com
+  name: <pv-name> <1>
+spec:
+  capacity:
+    storage: 10Gi <2>
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain  
+  storageClassName: <sc-name> <3>
+  mountOptions:
+    - cache=strict
+    - nosharesock
+    - actimeo=30
+    - nobrl 
+  csi:
+    driver: file.csi.azure.com
+    volumeHandle: "{resource-group-name}#{storage-account-name}#{file-share-name}" <4>
+    volumeAttributes:
+      shareName: <existing-file-share-name> <5>
+    nodeStageSecretRef:
+      name: <secret-name>  <6>
+      namespace: <secret-namespace>  <7>
+----
+<1> The name of the PV.
+<2> The size of the PV.
+<3> The storage class name.
```

Comment:
In case of static provisioning the SC name does not really matter it does not need to actually exists right? Should we call it out explicitly to avoid confusion ? i.e avoid customers wondering if they should create one.

@duanwei33

Reply:
I think it matters, the sc name in PV and PVC helps binding, it could help avoid unexpected behaviour. (Like there are other azure-disk PVs without sc name and the size is the same)
But yes, the storage class itself (with the same name) exists or not will not impact the binding, I think for customer scenario,  non-admin users have no access to list PV, so usually an admin will create a sc which a meaningful name and create pvs with the same sc name inside, then the non-admin users create the PVC with the sc for expected pvs.

In short, let's keep sc name in both pvc and pv.

Reply:
ok to keep it but it can be misleading as customers will not know what to set here.

Reply:
so.....leave as is or change?

Reply:
I personally think we should keep the SC in the example + comment that it's used to disable dynamic provisioning for such PV / PVC.

> What happens if a user sets an existing (unrelated) storage class?

In theory nothing, the PVC will try to bind to an existing PV. In practice, when the manually created PV and PVC accidentally don't match, Kubernetes will try to dynamically provision a new PV. Which is not what the user wants. Either a dedicated SC with `provisioner: kubernetes.io/no-provisioner` or non-existing SC will ensure the dynamic provisioning does not happen. Mis-matched PV + PVC stay Pending and user can fix their errors.

Reply:
Agree to keep the SC in the PVC/PV definition i was just wondering if we could add an extra note on the fact that the SC should not actually to avoid potential user errors (or as Jan said create a no-provisioner one). Not mentioning anything can be error prone i think

Reply:
excellent, way more clear for users now, thanks